pkg/ruler/api.go

```diff
@@ -119,6 +125,26 @@ func respondError(logger log.Logger, w http.ResponseWriter, msg string) {
 	}
 }
 
+func respondClientError(logger log.Logger, w http.ResponseWriter, msg string) {
+	b, err := json.Marshal(&response{
+		Status:    "error",
+		ErrorType: v1.ErrServer,
```

Comment:
If this function returns 4xx, then shouldn't the ErrorType be `v1.ErrClient`?

Reply:
Changed this to `v1.ErrBadData`